src/global_tracer.cpp

```diff
@@ -0,0 +1,26 @@
+
+#include "global_tracer.h"
+
+#include <datadog/tracer.h>
+
+#include <optional>
+
+namespace datadog {
+namespace nginx {
+namespace {
+std::optional<dd::Tracer> instance;
+}  // namespace
+
+dd::Tracer* global_tracer() {
+  if (instance) {
+    return &*instance;
+  }
+  return nullptr;
+}
+
+void reset_global_tracer() { instance.reset(); }
+
+void reset_global_tracer(dd::Tracer&& tracer) { instance = std::move(tracer); }
```

Comment:
Maybe this should be called set_global_tracer instead of overloading reset_global_tracer

Reply:
I prefer the symmetry, but your suggestion is tempting.

Reply:
There's no right answer and it really doesn't matter.
I generally read "reset" as setting things back to zero/null/initial-state instead of "re-set", setting it to something different (that could also be that initial-state)